`LD_FLAGS` may need to be updated for recent binutils versions.

Consider the following `binutils` version:

```
william@xubuntu-dtrain:~/Projects/FPGA/litex/litex-boards$ powerpc64le-linux-gnu-ld -v
GNU ld (GNU Binutils) 2.36
```

When compiling a litex CPU with a recent `binutils`, compiling the BIOS will fail because of an unknown command line argument:

```
python3 litex_boards/targets/orangecrab.py --cpu-type=microwatt --cpu-variant=standard+ghdl+irq --integrated-rom-size=0xa000
...
 LD       bios.elf
powerpc64le-linux-gnu-ld: Error: unable to disambiguate: -nodefaultlibs (did you mean --nodefaultlibs ?)
make: *** [/home/william/Projects/FPGA/litex/litex/litex/soc/software/bios/Makefile:67: bios.elf] Error 1
make: Leaving directory '/home/william/Projects/FPGA/litex/litex-boards/build/orangecrab/software/bios'
Traceback (most recent call last):
  File "litex_boards/targets/orangecrab.py", line 246, in <module>
    main()
  File "litex_boards/targets/orangecrab.py", line 239, in main
    builder.build(**builder_kargs, run=args.build)
  File "/home/william/Projects/FPGA/litex/litex/litex/soc/integration/builder.py", line 210, in build
    self._generate_rom_software(not self.soc.integrated_rom_initialized)
  File "/home/william/Projects/FPGA/litex/litex/litex/soc/integration/builder.py", line 190, in _generate_rom_software
    subprocess.check_call(["make", "-C", dst_dir, "-f", makefile])
  File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['make', '-C', '/home/william/Projects/FPGA/litex/litex-boards/build/orangecrab/software/bios', '-f', '/home/william/Projects/FPGA/litex/litex/litex/soc/software/bios/Makefile']' returned non-zero exit status 2.
```

It appears that `-nodefaultlibs` is not an argument `ld` understands (perhaps never was?), and binutils has gotten more strict with argument parsing recently.

The following patch fixes linking the BIOS for my use case, but I'm not sure how other CPUs are affected:

```diff
diff --git a/litex/soc/software/common.mak b/litex/soc/software/common.mak
index ec14f40d..754b26ef 100644
--- a/litex/soc/software/common.mak
+++ b/litex/soc/software/common.mak
@@ -53,7 +53,7 @@ INCLUDES = -I$(SOC_DIRECTORY)/software/include/base \
 COMMONFLAGS = $(DEPFLAGS) -Os $(CPUFLAGS) -g3 -fomit-frame-pointer -Wall -fno-builtin -nostdinc -fno-stack-protector $(INCLUDES)
 CFLAGS = $(COMMONFLAGS) -fexceptions -Wstrict-prototypes -Wold-style-definition -Wmissing-prototypes
 CXXFLAGS = $(COMMONFLAGS) -std=c++11 -I$(SOC_DIRECTORY)/software/include/basec++ -fexceptions -fno-rtti -ffreestanding
-LDFLAGS = -nostdlib -nodefaultlibs -L$(BUILDINC_DIRECTORY)
+LDFLAGS = -nostdlib -L$(BUILDINC_DIRECTORY)

 define compilexx
 $(CX) -c $(CXXFLAGS) $(1) $< -o $@
```

--- a/litex/soc/software/bios/Makefile
+++ b/litex/soc/software/bios/Makefile
@@ -64,7 +64,7 @@ bios.elf: $(BIOS_DIRECTORY)/linker.ld $(OBJECTS)
 	../libfatfs/libfatfs.a \
 	../liblitesdcard/liblitesdcard.a \
 	../liblitesata/liblitesata.a
-	$(LD) $(LDFLAGS) -T $(BIOS_DIRECTORY)/linker.ld -N -o $@ \
+	$(CC) $(LDFLAGS) -T $(BIOS_DIRECTORY)/linker.ld -N -o $@ \
 		../libbase/crt0.o \
 		$(OBJECTS) \
 		-L../libcompiler_rt \

--- a/litex/soc/software/demo/Makefile
+++ b/litex/soc/software/demo/Makefile
@@ -18,7 +18,7 @@ all: demo.bin
 	chmod -x $@
 
 demo.elf: $(OBJECTS)
-	$(LD) $(LDFLAGS) \
+	$(CC) $(LDFLAGS) \
 		-T linker.ld \
 		-N -o $@ \
 		$(BUILD_DIR)/software/libbase/crt0.o \

